### PR TITLE
Delay metrics deletion until after service has been deleted

### PR DESCRIPTION
**Pull request by shamsimam**

### CODE REVIEW

**Comment by shamsimam** [comment]
@sradack @dposada any feedback on this?

File: waiter/src/waiter/metrics.clj [hunk 1]
```diff
@@ -361,36 +361,57 @@
...
+                                     first
+                                     second
+                                     :available-apps
+                                     set)
+          service-id->state (pc/map-from-keys (fn service-id->state-fn [service-id]
+                                                (-> (get service-id->metrics service-id)
+                                                    (select-keys ["outstanding" "total"])
+                                                    (assoc "alive?" (contains? available-service-ids service-id))))
+                                              (keys service-id->metrics))]
+      (if (not (and (nil? service-id->metrics) (nil? scheduler-messages)))
```
Comment (sradack) [inline waiter/src/waiter/metrics.clj]: `(if (or service-id->metrics scheduler-messages)`

Comment (shamsimam) [inline waiter/src/waiter/metrics.clj]: Done.

File: waiter/src/waiter/metrics.clj [hunk 2]
```diff
@@ -361,36 +361,57 @@
...
+        (do
+          (async/>! service-id->state-chan service-id->state)
+          (recur))
+        (log/info "[transient-metrics-gc] stopping populating service-id->state-chan")))))
+
 (defn transient-metrics-gc
   "Launches go-blocks that keep running for the lifetime of the router watching for idle services known to the router.
    When such an idle service is found (based on the timestamp it was last updated), the metrics (except the one for
```
Comment (shamsimam) [inline waiter/src/waiter/metrics.clj]: Done.

**Pull request merged**